logical/plugin/plugin.go

```diff
@@ -0,0 +1,97 @@
+package plugin
+
+import (
+	"fmt"
+
+	"sync"
+
+	"github.com/hashicorp/go-plugin"
+	"github.com/hashicorp/vault/helper/pluginutil"
+	"github.com/hashicorp/vault/logical"
+)
+
+// PluginMap should be used by clients for map of plugins
+var PluginMap = map[string]plugin.Plugin{
```

Comment:
Right, this had to be exposed before I switched to using `pluginrunner.Run()`, but can be safely moved into `newPluginClient()` now.